### PR TITLE
IGNITE-10543: Test/train sample generator

### PR DESCRIPTION
**Pull request by avplatonov**

### CODE REVIEW

File: modules/ml/src/main/java/org/apache/ignite/ml/util/generators/primitives/scalar/UniformRandomProducer.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+     * Creates an instance of UniformRandomProducer.
+     *
+     * @param from Generate values from this value.
+     * @param to Generate values to this value.
+     * @param seed Seed.
+     */
+    public UniformRandomProducer(double from, double to, long seed) {
+        super(seed);
+
+        A.ensure(to >= from, "from > to");
```
Comment (dmitrievanthony) [inline modules/ml/src/main/java/org/apache/ignite/ml/util/generators/primitives/scalar/UniformRandomProducer.java]: Doesn't look like conditions are same.

File: modules/ml/src/main/java/org/apache/ignite/ml/util/generators/primitives/vector/VectorGeneratorPrimitives.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    /**
+     * Returns vector generator of vectors from multidimension gauss distribution.
+     *
+     * @param means Mean values per dimension.
+     * @param variances Variance values per dimension.
+     * @param seed Seed.
+     * @return Generator.
+     */
+    public static VectorGenerator gauss(Vector means, Vector variances, Long seed) {
+        A.notEmpty(means.asArray(), "mean.size() != 0");
```
Comment (dmitrievanthony) [inline modules/ml/src/main/java/org/apache/ignite/ml/util/generators/primitives/vector/VectorGeneratorPrimitives.java]: Do we actually need to convert vector into array just to check that it's not empty?

File: modules/ml/src/main/java/org/apache/ignite/ml/util/generators/primitives/vector/VectorGeneratorPrimitives.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+            .vectorize(2)
+            .filter(v -> Math.sqrt(v.getLengthSquared()) <= radius);
+    }
+
+    /**
+     * @param size Vector size.
+     * @return Generator of constant vector = zero.
+     */
+    public static VectorGenerator zero(int size) {
+        return constant(VectorUtils.of(new double[size]));
```
Comment (dmitrievanthony) [inline modules/ml/src/main/java/org/apache/ignite/ml/util/generators/primitives/vector/VectorGeneratorPrimitives.java]: I guess we could use `VectorUtils.zeroes(size)` here.

File: modules/ml/src/main/java/org/apache/ignite/ml/util/generators/primitives/scalar/DiscreteRandomProducer.java [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+
+    /**
+     * Creates a producer of random values from uniform discrete distribution.
+     *
+     * @param numberOfValues Number of distinct values.
+     * @param seed Seed.
+     * @return Producer.
+     */
+    public static DiscreteRandomProducer uniform(int numberOfValues, long seed) {
+        return new DiscreteRandomProducer(seed, IntStream.range(0, numberOfValues)
```
[the remark this answers is not in the document]
Comment (avplatonov) [reply modules/ml/src/main/java/org/apache/ignite/ml/util/generators/primitives/scalar/DiscreteRandomProducer.java]: yes, you're right

File: modules/ml/src/main/java/org/apache/ignite/ml/util/generators/primitives/vector/VectorGeneratorPrimitives.java (hunk 3 above)
Comment (avplatonov) [reply modules/ml/src/main/java/org/apache/ignite/ml/util/generators/primitives/vector/VectorGeneratorPrimitives.java]: thank you

**Pull request closed without merging**